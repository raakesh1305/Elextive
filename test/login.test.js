import Component from '../pages/auth/login.vue'
import Login from '../pages/auth/login.vue';
import {shallow, mount, shallowMount } from '@vue/test-utils';
describe('Component', () => {


  test('is a Vue instance', () => {
    const wrapper = mount(Component)
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
}
,
test('renders correctly', () => {
    const wrapper = mount(Component)
    expect(wrapper.element).toMatchSnapshot()
  })
)

describe('Login Component', () => {
 
 // make our assertion and what we expect to happen 
 it('should render without throwing an error', () => {
   expect(shallow(Login ).find('.error').exists()).toBe(false)
 })
})

describe("Login has a button", () => {
  

  test("Has a button", () => {
    const wrapper = shallowMount(Component);
    expect(wrapper.contains("button")).toBe(true);
  });

});

