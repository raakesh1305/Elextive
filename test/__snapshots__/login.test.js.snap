// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="hero is-fullheight"
>
  <div
    class="columns"
  >
    <div
      class="column is-4 banner"
    >
      <h1
        class="is-size-1 has-text-centered"
      >
        Login
      </h1>
       
      <p
        class="content has-text-centered is-size-5 is-hidden-mobile"
      >
        
        Welcome back to Julius ðŸ‘‹
      </p>
    </div>
     
    <div
      class="column form"
    >
      <label
        class="label"
      >
        Your Credentials:
      </label>
       
      <div
        class="field"
      >
        <div
          class="control"
        >
          <input
            class="input"
            placeholder="Username/Email "
            type="text"
          />
        </div>
      </div>
       
      <div
        class="field"
      >
        <div
          class="control"
        >
          <input
            class="input"
            placeholder="Password"
            type="password"
          />
        </div>
      </div>
       
      <div
        class="field is-grouped"
      >
        <div
          class="control"
        >
          <button
            class="button is-link is-light is-rounded"
          >
            Login
            â†’
          </button>
        </div>
         
        <div
          class="control"
        >
          <nuxt-link
            class="button is-danger is-light is-rounded"
            to="/"
          >
            
            Cancel
          
          </nuxt-link>
        </div>
         
        <div
          class="control float-right"
        >
          <nuxt-link
            class="button is-light is-warning is-outlined is-rounded"
            to="/auth/reset"
          >
            Forgot Password?
          </nuxt-link>
        </div>
      </div>
    </div>
  </div>
   
  <modal
    classes="modal is-active"
    name="error"
  >
    <div
      class="modal-background"
    />
     
    <div
      class="modal-content"
    >
      <div
        class="section"
      >
        <h3
          class="title has-text-white"
        >
          Oh no!
        </h3>
         
        <p
          class="subtitle is-size-4 has-text-white"
        >
          
        </p>
         
        <div
          class="buttons"
        >
          <button
            class="button is-danger is-rounded"
          >
            Close
          </button>
        </div>
      </div>
    </div>
  </modal>
</div>
`;
